### PR TITLE
Fetch chain, version, ss58Prefix from api

### PR DESCRIPTION
**Pull request by TingALin**
issue #1173

### CODE REVIEW

File: src/contexts/Api/index.tsx [hunk 1]
```diff
@@ -114,6 +115,9 @@ export const APIProvider = ({ children }: { children: React.ReactNode }) => {
   // Store network constants.
   const [consts, setConsts] = useState<APIConstants>(defaults.consts);
 
+  // Store network meta data.
+  const [meta, setMeta] = useState<APIMeta>(defaults.meta);
```
Comment (rossbulat) [inline src/contexts/Api/index.tsx:119]: maybe call this chainState

File: src/contexts/Api/index.tsx [hunk 2]
```diff
@@ -184,6 +196,7 @@ export const APIProvider = ({ children }: { children: React.ReactNode }) => {
       newApi.consts.staking.historyDepth,
```
Comment (rossbulat) [inline src/contexts/Api/index.tsx:196]: separate chainState and consts into 2 separate functions, & fetch consts in a useEffect once chainState has been fetched.

File: src/contexts/Api/types.ts [hunk 3]
```diff
@@ -24,12 +24,19 @@ export interface APIConstants {
   existentialDeposit: BigNumber;
   fastUnstakeDeposit: BigNumber;
   poolsPalletId: U8aLike;
+  ss58Prefix: BigNumber;
+}
+
+export interface APIChainState {
+  chain: string;
+  version: string;
```
Comment (rossbulat) [inline src/contexts/Api/types.ts:32]: We could also make these optional but I don't that that would be correct; we always need to know if a chain is assigned or not.

```suggestion
  chain: string | undefined;
  version: string | undefined;
```

File: src/contexts/Api/index.tsx [hunk 4]
```diff
@@ -158,12 +164,32 @@ export const APIProvider = ({ children }: { children: React.ReactNode }) => {
       provider.on('error', () => {
         setApiStatus('disconnected');
       });
-      connectedCallback(provider);
+      connectedChainStateCallback(provider);
+      connectedConstsCallback(provider);
```
[the remark this answers is not in the document]
Comment (rossbulat) [reply src/contexts/Api/index.tsx:168]: Like we discussed, once the chain state is fetched, then the consts can be fetched. Right now you are fetching them both together.

[the remark this answers is not in the document]
Comment (TingALin) [reply src/contexts/Api/index.tsx:168]: Using the same dependency? May I ask why?

File: src/contexts/Api/defaults.ts [hunk 5]
```diff
@@ -18,6 +22,12 @@ export const consts: APIConstants = {
   existentialDeposit: new BigNumber(0),
   fastUnstakeDeposit: new BigNumber(0),
   poolsPalletId: stringToU8a('0'),
+  ss58Prefix: new BigNumber(0),
+};
+
+export const chainState: APIChainState = {
+  chain: '',
+  version: '',
```
[the remark this answers is not in the document]
Comment (rossbulat) [reply src/contexts/Api/defaults.ts:30]: Let's just focus on the `API` context here, I can't see any more empty strings in this context. We can explore improving the types for other contexts in other PRs.

the benefit is that there is less ambiguity by using an undefined value. An empty string could be mistaken as an assigned value when in fact it means that the value has not yet been assigned.

File: src/contexts/Api/index.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (rossbulat) [reply src/contexts/Api/index.tsx:168]: The dependencies will be different. We are separating these steps as you only want to fetch the consts once we can confirm that the correct chain and version has been connected to

[the remark this answers is not in the document]
Comment (rossbulat) [reply src/contexts/Api/index.tsx:168]: The dependencies will be different. We are separating these steps as you only want to fetch the consts once we can confirm that the correct chain and version has been connected to

**Pull request closed without merging**